executor/distsql.go

```diff
@@ -846,11 +855,22 @@ func (w *tableWorker) pickAndExecTask(ctx context.Context) {
 	}
 }
 
-func (e *IndexLookUpExecutor) getHandle(row chunk.Row, handleIdx []int, isCommonHandle bool) (handle kv.Handle, err error) {
+func (e *IndexLookUpExecutor) getHandle(row chunk.Row, handleIdx []int,
+	isCommonHandle bool, tp getHandleType) (handle kv.Handle, err error) {
 	if isCommonHandle {
 		var handleEncoded []byte
 		var datums []types.Datum
 		for i, idx := range handleIdx {
+			// If the new collation is enabled and the handle contains non-binary string,
+			// we change the collation to "utf8mb4_bin" to avoid encode sortKey again.
```

Comment:
```suggestion
			// we change the collation to "binary" to avoid encode sortKey again.
```
BTW, I'm not sure why `binary` and `utf8mb4_bin` both works here.